src/classes/WalletManager/eip1193Provider.ts

```diff
@@ -3,15 +3,9 @@ export interface RequestArguments {
     params?: unknown[] | Record<string, unknown>
 }
 
-export interface MetaMaskEthereumProvider {
-    isMetaMask?: boolean
-
-    once(eventName: string | symbol, listener: (...args: any[]) => void): this
-
+export interface EIP1193Provider {
     on(eventName: string | symbol, listener: (...args: any[]) => void): this
 
-    off(eventName: string | symbol, listener: (...args: any[]) => void): this
-
     addListener(eventName: string | symbol, listener: (...args: any[]) => void): this
 
     removeListener(eventName: string | symbol, listener: (...args: any[]) => void): this
```

Comment:
are these functions optional or required?

Reply:
Currently, they are required, since the ethereum provider interface extends a Nodejs event emitter, all providers "should" have those methods. But we can leave only the ones that we actually use :thinking: ...